lib/matplotlib/finance.py

```diff
@@ -1139,8 +1139,8 @@ def candlestick2_ohlc(ax, opens, highs, lows, closes, width=4,
     ax.autoscale_view()
 
     # add these last
-    ax.add_collection(barCollection)
     ax.add_collection(rangeCollection)
+    ax.add_collection(barCollection)
```

Comment:
What is the reason for swapping the order here?  I wouldn't expect it to make any difference, since the difference in zorder should determine the actual drawing order.